go-controller/pkg/ovn/egressgw.go

```diff
@@ -98,10 +98,7 @@ func (oc *Controller) addPodExternalGWForNamespace(namespace string, pod *kapi.P
 	}
 	klog.Infof("Adding routes for external gateway pod: %s, next hops: %q, namespace: %s, bfd-enabled: %t",
 		pod.Name, gws, namespace, egress.bfdEnabled)
-	nsInfo, err := oc.waitForNamespaceLocked(namespace)
-	if err != nil {
-		return err
-	}
+	nsInfo := oc.ensureNamespaceLocked(namespace)
```

Comment:
nit: maybe just put a comment here saying something like "no need to call ensureNamespaceLocked since it was already called in the pod creation workflow" 

It took me a second to look through the code and to confirm this shouldn't be `ensureNamespaceLocked`

Reply:
yeah really ensureNamespaceLocked only needs to be called in the handlers first iteration of where it needs the namespace to exist. It's not obvious where that is without knowing the handlers. addPodExternalGWForNamespace can be called from outside of addLogicalPort if it is a host networked pod, so this needs to be ensure. Really you could use ensure in most cases, but semantically its probably easier to understand if we keep it separate and where we use getNamsepaceLocked indicates to the reader that the namespace should already exist by now.